extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js

```diff
@@ -59,25 +60,37 @@ require([
 		});
 	}
 
+	function formatAgoString(unixTime, locale) {
+		var date = moment.unix(unixTime),
+			now = moment();
+
+		if (now.diff(date, 'days') > 5) {
+			return date.locale(locale).format('L');
+		} else if (now.diff(date, 'minutes') < 1) {
+			return $.msg('embeddable-discussions-timestamp-now');
+		} else {
+			return date.locale(locale).fromNow();
+		}
+	}
+
 	function processData(threads, upvoteUrl) {
 		var ret = [],
 			i,
 			thread,
 			userData,
-			date;
+			locale = mw.config.get('wgContentLanguage');
 
 		for (i in threads) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)